src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -323,14 +332,7 @@ private void ApplySettings(ConsumerSettings<K, V> updatedSettings)
 
                 var localSelf = Self;
                 _consumer = _settings.CreateKafkaConsumer(
-                    consumeErrorHandler: (c, e) =>
```

Comment:
Interesting - so we just deliver a message to ourselves and handle it inside the actor's scope rather than externally now? Seems like a good idea, but why this change? Is it just to make sure that the decider is "in charge?"

Reply:
Yes, the consumeErrorHandler callback is called on internal errors like unknown topic or partition. Not all errors are propagated to the `Consume` method. By routing this error to the actor, error handling become a lot more consistent, Plus we avoid any possible threading problems.